### PR TITLE
[Interpreter] Implement objects and strings

### PR DESCRIPTION
**Pull request by dkofanov**
Depends on https://github.com/K3S-prod/k3s-virtual-machine/pull/22

### CODE REVIEW

File: assembler/assembler.h [hunk 1]
```diff
@@ -147,21 +183,37 @@ class AsmEncoder {
         if (ENCODER.unresolved_labels_.size() != 0) {
             LOG_FATAL(ENCODER, "There are " << ENCODER.unresolved_labels_.size() << " unresolved labels!");
         }
+        if (ENCODER.instructions_buffer_.back().GetOpcode() != Opcode::RET) {
+            LOG_FATAL(ENCODER, "Function should end with `RET`");
+        }
```
Comment (Dimanteo) [inline assembler/assembler.h:188]: Are we going to forbid contructions such as
```
FuncEntry:
    b Lbl1
Exit:
    ret
Lbl1:
    b Exit
```

Comment (dkofanov) [reply assembler/assembler.h:188]: No, this is a temporary solution (consider it as assembler restriction).

**Review by Dimanteo — APPROVED** [review]

**Review by Dimanteo — APPROVED** [review]

File: classfile/class_file.cpp [hunk 2]
```diff
@@ -168,28 +204,65 @@ BytecodeInstruction *ClassFile::LoadCodeSection(void *filebuf)
...
+            for (size_t i = 0; i < record->data_fields_n_; i++) {
+                const char *c_str = reinterpret_cast<const char *>(constpool_file + pos);
+                if (mapping->find(c_str) != mapping->end()) {
+                    LOG_FATAL(INTERPRETER, "Object has overlapping fields names");
+                }
+                (*mapping)[c_str] = i;
+                while (*c_str != '\0') {
+                    pos++;
+                    c_str = reinterpret_cast<const char *>(constpool_file + pos);
+                }
```
Comment (Dimanteo) [inline classfile/class_file.cpp:243]: Idea for future refactoring - can use `strlen()` here

Comment (Dimanteo) [reply classfile/class_file.cpp:243]:
```suggestion
                pos += strlen(c_str);
                c_str = reinterpret_cast<const char *>(constpool_file + pos);
```

File: classfile/class_file.cpp [hunk 3]
```diff
@@ -168,28 +204,65 @@ BytecodeInstruction *ClassFile::LoadCodeSection(void *filebuf)
...
+            for (size_t i = 0; i < record->methods_n_; i++) {
+                const char *c_str = reinterpret_cast<const char *>(constpool_file + pos);
+                if (mapping->find(c_str) != mapping->end()) {
+                    LOG_FATAL(INTERPRETER, "Object has overlapping fields names");
+                }
+                (*mapping)[c_str] = record->data_fields_n_ + i;
+                while (*c_str != '\0') {
+                    pos++;
+                    c_str = reinterpret_cast<const char *>(constpool_file + pos);
+                }
```
Comment (Dimanteo) [inline classfile/class_file.cpp:255]: Same thing with `strlen()` here

File: classfile/class_file.cpp (hunk 2 above)
Comment (dkofanov) [reply classfile/class_file.cpp]: Applied, thanks.

**Pull request merged**